torch/distributed/distributed_c10d.py

```diff
@@ -883,8 +962,8 @@ def _new_process_group_helper(
                         world_size=group_size,
                         timeout=timeout,
                     )
-            _pg_map[pg] = (Backend.GLOO, store)
```

Comment:
FYI I have a PR to refactor some of this code since this and the if statements below are basically repeated, https://github.com/pytorch/pytorch/pull/88351. Feel free to land yours first if you are ready though